File: prototypes/angular-node-javascript/backend-node-javascript-postgresql/src/features/person/person-service.js
class PersonService {
  constructor(repository) {
    this.repository = repository;
  }

  async getItems() {
    return this.repository.getItems();
  }

  async getItemById(id) {
    return this.repository.getItemById(id);
  }

  async createItem(person) {
    return this.repository.createItem(person);
  }

  async updateItem(id, updatedData) {
    return this.repository.updateItem(id, updatedData);
  }

  async deleteItem(id) {
    return this.repository.deleteItem(id);
  }
}

export default PersonService;

File: mcp/prototypes/002-api-rest/backend-springboot/src/main/java/com/ganatan/modules/person/PersonService.java
package com.ganatan.modules.person;

import org.springframework.stereotype.Service;
import java.util.*;

@Service
public class PersonService {

    private final Map<Long, Person> repository = new HashMap<>();

    public PersonService() {
        repository.put(1L, new Person(1L, "Steven Spielberg"));
        repository.put(2L, new Person(2L, "Martin Scorsese"));
    }

    public List<Person> getItems() {
        return new ArrayList<>(repository.values());
    }

    public Person getItemById(Long id) {
        return repository.get(id);
    }

    public Person createItem(Person person) {
        if (repository.containsKey(person.getId())) {
            throw new IllegalStateException("ALREADY_EXISTS");
        }
        repository.put(person.getId(), person);
        return person;
    }

    public Person updateItem(Long id, Person newPerson) {
        if (!repository.containsKey(id)) {
            return null;
        }
        repository.put(id, new Person(id, newPerson.getName()));
        return repository.get(id);
    }

    public boolean deleteItem(Long id) {
        return repository.remove(id) != null;
    }
}

